docs/setup-project.adoc

```diff
@@ -0,0 +1,47 @@
+
+Hercules CI is configured "by convention" using `.nix` files in your repository.
+Follow these steps to ensure a correct setup.
+
+// TODO: If you haven't set up a project with Nix before, we recommend the <<tutorial>>
+
+1. Clone your repository
+
+2. Choose which Nix file to use for CI
++
+Reuse `/default.nix`::
+For simple projects that only need to build for a single architecture and a single set of dependencies.
+or create a `/ci.nix` or `/nix/ci.nix`::
+For projects that need to build for multiple architectures or versions, or where `default.nix` already
+has a special meaning that conflicts with the need for CI.
+
+3. Use pinned dependencies instead of `<nixpkgs>` or similar.
++
+Hercules CI uses an empty `NIX_PATH` to help with evaluation reprodicibility.
++
+The unofficial NixOS wiki has https://nixos.wiki/wiki/FAQ/Pinning_Nixpkgs[a basic method of pinning Nixpkgs].
++
+You can use an impure evaluation-time fetcher such as `builtins.fetchUrl`, although this typically makes your evaluation unreproducible.
```

Comment:
```suggestion
```

Why would we recommend this? I can't think of a reason.

Reply:
Not sure how harmful it is. It does make setup easier. Happy to remove if you stick to it.

Reply:
The guide shouldn't offer options I think. We should recommend one way and then link to choice into reference if needed.

Using `builtins.fetchTarball` suffices to bootstrap but this will be soon gone with flakes anyway.